packages/apputils/src/dialog.ts

```diff
@@ -65,21 +62,29 @@ class Dialog extends Widget {
     this._buttonNodes = toArray(map(this._buttons, button => {
       return renderer.createButtonNode(button);
     }));
-    this._primary = (
-      options.primaryElement || this._buttonNodes[this._defaultButton]
-    );
 
     let layout = this.layout = new PanelLayout();
     let content = new Panel();
     content.addClass('jp-Dialog-content');
     layout.addWidget(content);
 
+    this._body = options.body;
+
     let header = renderer.createHeader(options.title);
     let body = renderer.createBody(options.body);
     let footer = renderer.createFooter(this._buttonNodes);
     content.addWidget(header);
     content.addWidget(body);
     content.addWidget(footer);
+
+    this._primary = this._buttonNodes[this._defaultButton];
+
+    if (options.primaryElement) {
```

Comment:
Not a fan of "primaryElement" name, when its actually a query string. Something more descriptive like "focusNodeSelector" or "focusQuerySelector" or "primaryElementSelector" would be better.